### PR TITLE
Add metrics and internal command for AutoRecovery

### PR DESCRIPTION
**Pull request by frankxieke**
Motivation:

Current AutoRecovery does not have enough metrics or stat command that would help to monitor and debug. So we need to add metrics and admin stat interface to monitor the process of AutoRecovery.  For example, current recovering ledgerInfo and under replicated size, read/write latency in recovering.

Changes:

And QueryAutoRecoveryStatus command and under replicated size metric , read/write latency metric in recovering

Documentation:

Need doc.

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java [hunk 1]
```diff
@@ -310,6 +327,8 @@ private void recoverLedgerFragmentEntry(final Long entryId,
         final long ledgerId = lh.getId();
         final AtomicInteger numCompleted = new AtomicInteger(0);
         final AtomicBoolean completed = new AtomicBoolean(false);
+        Stopwatch readEntryWatcher = Stopwatch.createUnstarted();
```
[the remark this answers is not in the document]
Comment (frankxieke) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java]: ok, i will fix it.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/autorecovery/QueryAutoRecoveryStatusCommand.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+     CurrentRecoverLedgerInfo:
+        LedgerId:   BookieId:     LedgerSize:(detail)
+        LedgerId:   BookieId:     LedgerSize:(detail)
+     */
+    public boolean handler(ServerConfiguration conf, QFlags flag) throws Exception {
+        ZooKeeper zk = null;
+        try {
+            String metadataServiceUri = conf.getMetadataServiceUri();
+            String zkServers = ZKMetadataDriverBase.getZKServersFromServiceUri(URI.create(metadataServiceUri));
+            zk = ZooKeeperClient.newBuilder()
```
[the remark this answers is not in the document]
Comment (frankxieke) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/autorecovery/QueryAutoRecoveryStatusCommand.java]: I have fixed it. Now i use the MetadataDriver abstraction interface to solve this problem. Please feel free to ping me if you any suggestion. thanks.

**Comment by hangc0276** [comment]
rerun failure checks

**Comment by frankxieke** [comment]
rerun failure checks

**Comment by frankxieke** [comment]
rerun failure checks

**Comment by frankxieke** [comment]
rerun failure checks

**Comment by frankxieke** [comment]
@eolivelli Hi, Could you have a look at this PR. Thanks

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**